### PR TITLE
Fixes Email Validation

### LINKED ISSUE #1996: Add Email Validation
Hi, this is a first-timers-only issue. This means we've worked to make it more legible to folks who either haven't contributed to our codebase before, or even folks who haven't contributed to open source before.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you :gift_heart:

If you have contributed before, consider leaving this one for someone new, and looking through our general help wanted issues. Thanks!
:thinking: What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.
## Please describe the problem

A user is able to sign in and use publiclab with email wrong email addresses say bansal@gmail

:clipboard: Step by Step

:raising_hand_woman: Claim this issue: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

:memo: Add VALID_EMAIL_REGEX = /\A[\w+-.]+@[a-z\d-.]+.[a-z]+\z/i to the 
https://github.com/publiclab/plots2/blob/3b09a3515bbfecb48194bd86f2d467b2710103be/app/models/user.rb#L16
:floppy_disk: Commit your changes

:twisted_rightwards_arrows: Start a Pull Request.

:checkered_flag: Done Ask in comments for a review :)

:thinking::question: Questions?

Leave a comment below!

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. If someone seem stuck, offer them some help! Otherwise, take a look at some other issues you can help with. Thanks!
Thank you!

Your help makes Public Lab better! We deeply appreciate your helping refine and improve this site.

### PR DESCRIPTION
**Pull request by namanjain1812**
Closes #1996 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by namanjain1812** [comment]
@SidharthBansal Please merge it and thanks for all the help.
will like to contribute more in the future  😃

**Comment by SidharthBansal** [comment]
@namanjain1812 can you please update regex to  /\A[-[:alnum:]+.]+@[[:alnum:]-.]+[.][[:alpha:]]+\z/ as we discussed in #2169 so that it will include the non-latin characters? 
Also can you please add tests to check that validation errors are properly working.

**Comment by namanjain1812** [comment]
@SidharthBansal please review it.Also I have included 2 tests but i also tried adding a test including non-latin(chinese) words but Travis seems to not recognize them and hence gave checks failed message

**Comment by SidharthBansal** [comment]
@namanjain1812 It looks great.
We can omit the test for non latin characters if they are making a problem with travis. What do you suggest @jywarren ?

File: test/unit/user_test.rb [hunk 1]
```diff
@@ -178,5 +178,18 @@ class UserTest < ActiveSupport::TestCase
     assert_not basic_user.moderator?
     assert_not basic_user.can_moderate?
   end
-
+  test 'user email validation' do
+    user = User.new(username: 'zen',
+                    password: 'nez',
+                    password_confirmation: 'nez',
+                    email: 'abc@xyz.com')
```
Comment (seafr) [inline test/unit/user_test.rb]: This email is valid and the user should be saved to the database. 
What do you think @SidharthBansal?

**Comment by seafr** [comment]
@SidharthBansal I agree with you regarding the tests for emails with non-latin characters. This is a FTO issue, and we can leave such tests for a subsequent PR.

File: test/unit/user_test.rb (hunk 1 above)
Comment (SidharthBansal) [reply test/unit/user_test.rb]: Sorry I missed, you need to write it should not save to the database.
@seafr thanks

File: test/unit/user_test.rb [hunk 2]
```diff
@@ -182,7 +182,7 @@ class UserTest < ActiveSupport::TestCase
     user = User.new(username: 'zen',
                     password: 'nez',
                     password_confirmation: 'nez',
-                    email: 'abc@xyz.com')
+                    email: 'abc@.com')
```
Comment (seafr) [inline test/unit/user_test.rb]: That's good.

**Comment by jywarren** [comment]
Awesome -- is this good to go, then? Thanks, everybody!!! 👍 👍

**Comment by seafr** [comment]
Yes, I think so. The 'ready' label was removed due to that one test, which has since been fixed.

**Comment by jywarren** [comment]
Done, then!!! 🐯 👍 💯

**Comment by jywarren** [comment]
Thanks everybody!!!!

**Comment by seafr** [comment]
@namanjain1812 congrats on your first PR and welcome to the community!

**Pull request merged**